src/Shared/AssemblyNameExtension.cs

```diff
@@ -993,5 +1004,237 @@ public void GetObjectData(SerializationInfo info, StreamingContext context)
             info.AddValue("immutable", immutable);
             info.AddValue("remapped", remappedFrom);
         }
+
+#if !NET35
+        internal class Converter : JsonConverter<AssemblyNameExtension>
+        {
+            public override AssemblyNameExtension Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+            {
+                AssemblyNameExtension ane = new AssemblyNameExtension();
+                if (reader.TokenType != JsonTokenType.StartObject)
+                {
+                    throw new JsonException();
+                }
+                while (reader.Read())
+                {
+                    if (reader.TokenType == JsonTokenType.EndObject)
+                    {
+                        return ane;
+                    }
+                    else if (reader.TokenType == JsonTokenType.Null)
+                    {
+                        return null;
+                    }
+                    else if (reader.TokenType != JsonTokenType.PropertyName)
+                    {
+                        throw new JsonException();
+                    }
+                    string parameter = reader.GetString();
+                    reader.Read();
+                    if (reader.TokenType == JsonTokenType.Null)
```

Comment:
What does it mean when we see null here? What would we have just read from `GetString()` if we fall into this case?